src/rpc_api/data_types.rs

```diff
@@ -299,19 +299,38 @@ impl HasLotusJson for ActorState {
         )
     }
 }
-#[derive(Serialize, Deserialize, PartialEq)]
+#[derive(Serialize, Deserialize, PartialEq, Clone)]
 #[serde(rename_all = "PascalCase")]
 pub struct ApiActorState {
     #[serde(with = "crate::lotus_json")]
     balance: TokenAmount,
     #[serde(with = "crate::lotus_json")]
     code: Cid,
     #[serde(with = "crate::lotus_json")]
+    state: ApiState,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "PascalCase")]
+struct ApiState {
```

Comment:
Why hoist this out as a separate type? Why not directly use `Ipld`?

Reply:
this is to match lotus repsonse 
```
Object {"Balance": String("0"), "Code": Object {"/": String("bafk2bzacecioupndtcnyw6iq2hbrxag3aufvczlv5nobnfbkbywqzcyfaa376")}, "State": Object {"BuiltinActors": Object {"/": String("bafy2bzacecfgf3634uaq7lynhauelvpz77ebgouejdx6eqnokajdknylv4m6c")}}}
```

Reply:
Hmm, quite surprising that it has a `BuiltinActors` field... I think the Lotus RPC endpoint is buggy. We'll just have to copy those bugs.